connectors/sources/tests/test_confluence.py

```diff
@@ -0,0 +1,151 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+"""Tests the Confluence database source class methods"""
+import ssl
+from unittest.mock import patch
+
+import pytest
+
+from connectors.source import DataSourceConfiguration
+from connectors.sources.confluence import ConfluenceDataSource
+from connectors.sources.tests.support import create_source
+from connectors.utils import ssl_context
+
+HOST_URL = "http://127.0.0.1:5000"
+
+
+@pytest.mark.asyncio
+async def test_validate_configuration_with_invalid_concurrent_downloads():
+    """Test validate configuration method of BaseDataSource class with invalid concurrent downloads"""
+
+    # Setup
+    source = create_source(ConfluenceDataSource)
+    source.concurrent_downloads = 1000
+
+    # Execute
+    with pytest.raises(
+        Exception, match="Configured concurrent downloads can't be set more than *"
+    ):
+        await source.validate_config()
+
+
+def test_tweak_bulk_options():
+    """Test tweak_bulk_options method of BaseDataSource class"""
+
+    # Setup
+    source = create_source(ConfluenceDataSource)
+    options = {"concurrent_downloads": 10}
+
+    # Execute
+    source.tweak_bulk_options(options)
```

Comment:
What do we test here? That there's no exception or should we assert that options is actually mutated?

Reply:
updated with commit id [20563b6](https://github.com/elastic/connectors-python/pull/568/commits/20563b6738b72f418d41e48a130cc23d1537a631)